analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,249 @@
+        override val globsForDefinitionFiles = listOf("Cartfile*.resolved")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Carthage(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun resolveDependencies(definitionFile: File): List<ProjectAnalyzerResult> {
+        // Transitive dependencies are only supported if the dependency itself uses Carthage.
+        // See: https://github.com/Carthage/Carthage#nested-dependencies
+        val workingDir = definitionFile.parentFile
+        val projectInfo = getProjectInfoFromVcs(workingDir)
+
+        return listOf(
+            ProjectAnalyzerResult(
+                project = Project(
+                    id = Identifier(
+                        type = managerName,
+                        namespace = projectInfo.user.orEmpty(),
+                        name = projectInfo.project.orEmpty(),
+                        version = projectInfo.revision.orEmpty()
+                    ),
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    declaredLicenses = sortedSetOf(),
+                    vcs = VcsInfo.EMPTY,
+                    vcsProcessed = processProjectVcs(workingDir, VcsInfo.EMPTY),
+                    scopes = sortedSetOf(),
+                    homepageUrl = ""
+                ),
+                packages = parseCarthageDependencies(definitionFile)
+            )
+        )
+    }
+
+    /**
+     * As the "Carthage.resolved" file does not provide any project information, trying to retrieve some from VCS.
+     */
+    private fun getProjectInfoFromVcs(workingDir: File): ProjectInfo {
+        val workingTree = VersionControlSystem.forDirectory(workingDir)
+        val vcsInfo = workingTree?.getInfo() ?: VcsInfo.EMPTY
+        val normalizedVcsUrl = normalizeVcsUrl(vcsInfo.url)
+        val vcsHost = VcsHost.toVcsHost(URI(normalizedVcsUrl))
+
+        return ProjectInfo(
+            user = vcsHost?.getUserOrOrganization(normalizedVcsUrl),
+            project = vcsHost?.getProject(normalizedVcsUrl),
+            revision = vcsInfo.resolvedRevision
+        )
+    }
+
+    private fun parseCarthageDependencies(definitionFile: File): SortedSet<Package> {
+        val dependencyLines = definitionFile.readLines()
+        val workingDir = definitionFile.parent
+        val packages = sortedSetOf<Package>()
+
+        dependencyLines.forEach { line ->
+            if (line.isBlank() || line.isComment()) return@forEach
+            packages += parseDependencyLine(line, workingDir)
+        }
+
+        return packages
+    }
+
+    private fun parseDependencyLine(line: String, workingDir: String): Package {
+        val split = line.split(" ", limit = 3)
+        val type = DependencyType.valueOf(split[0].toUpperCase())
```

Comment:
If we don't have error handling when split.size != 3 here, should we at least make the assumption explicit that we have not only at most 3 elements, but exactly 3 elements? Like

    require(split.size == 3) {
        "A dependency line must consist of exactly 3 space separated elements."
    }

Reply:
Good catch. Implemented and added a unit test.